sycl/include/CL/sycl/types.hpp

```diff
@@ -542,12 +542,13 @@ using vec_data_t = typename detail::vec_helper<T>::RetType;
 // For information on calling conventions for x64 processors, see
 // Calling Convention
 // (https://docs.microsoft.com/en-us/cpp/build/x64-calling-convention).
-#pragma message ("Alignment of class vec is not in accordance with SYCL \
+#pragma message("Alignment of class vec is not in accordance with SYCL \
 specification requirements, a limitation of the MSVC compiler(Error C2719).\
-Applied default alignment.")
-#define __SYCL_ALIGNAS(x)
+Requested alignment applied, limited at 64.")
```

Comment:
@alexbatashev 
If I remember correctly you found that such a change is an ABI breaking one, could you please confirm?

Reply:
It is

Reply:
Could you please remind why?

Reply:
Type alignment is part of ABI: assembled code has no notion of structure layout, it operates raw bytes.

Reply:
In this particular change I do not see how setting alignment can break anything:
1. The vector type is not passed directly to functions, but as part of the struct, so it's either passed by pointer or thru a register.
2. The structure has only one field which is already considered to be aligned by the clang as it's a vector type(generates vector instructions)
3. We set the alignment of vector type being equal to the size of the field itself, so we should not change the size. of the structure to trigger it to be passed differently.